src/nvim/os/unix_defs.h

```diff
@@ -4,4 +4,75 @@
 #define TEMP_DIR_NAMES {"$TMPDIR", "/tmp", ".", "~"}
 #define TEMP_FILE_PATH_MAXLEN 256
 
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#endif
+
+#ifdef HAVE_SYS_PARAM_H
+# include <sys/param.h>     /* defines BSD, if it's a BSD system */
+#endif
+
+/* List 3 arg systems here. I guess __sgi, please test and correct me. jw. */
+#ifdef SIGHAS3ARGS
+#  define SIGDEFARG(s)  (int s, int sig2, struct sigcontext *scont)
+#else
+#  define SIGDEFARG(s)  (int s)
+#endif
+
+#ifndef MAXNAMLEN
+#  ifdef DIRSIZ
+#    define MAXNAMLEN DIRSIZ
+#  endif
+#  ifdef UFS_MAXNAMLEN
+#   define MAXNAMLEN UFS_MAXNAMLEN    /* for dynix/ptx */
+#  endif
+#endif
+
+#ifndef USR_EXRC_FILE
+#  define USR_EXRC_FILE "~/.exrc"
+#endif
+
+#ifndef USR_VIMRC_FILE
+#  define USR_VIMRC_FILE "~/.nvimrc"
+#endif
+
+#if !defined(USR_EXRC_FILE2)
+#    define USR_VIMRC_FILE2     "~/.nvim/nvimrc"
+#endif
+
+# ifndef VIMINFO_FILE
+#   define VIMINFO_FILE "~/.nviminfo"
+# endif
+
+#ifndef EXRC_FILE
+# define EXRC_FILE      ".exrc"
+#endif
+
+#ifndef VIMRC_FILE
+# define VIMRC_FILE     ".nvimrc"
+#endif
+
+#ifndef DFLT_BDIR
+#   define DFLT_BDIR    ".,~/tmp,~/"    /* default for 'backupdir' */
```

Comment:
The indentation should be consistent for `if(n)def`s (the style guide says two spaces, not sure it applies to pre-processor directives...)